conjure-go-contract/errors/generic_error.go

```diff
@@ -24,14 +24,17 @@ import (
 
 	"github.com/palantir/conjure-go-runtime/v2/conjure-go-contract/codecs"
 	"github.com/palantir/pkg/uuid"
+	werror "github.com/palantir/witchcraft-go-error"
 	wparams "github.com/palantir/witchcraft-go-params"
 )
 
-func newGenericError(errorType ErrorType, params wparams.ParamStorer) genericError {
+func newGenericError(cause error, errorType ErrorType, params wparams.ParamStorer) genericError {
 	return genericError{
 		errorType:       errorType,
 		errorInstanceID: uuid.NewUUID(),
 		params:          params,
+		cause:           cause,
+		stack:           werror.NewStackTrace(),
```

Comment:
Should we skip some frames so the error library isn't the top of every stacktrace?

Reply:
Yes, good catch. I adjusted this to skip 2 stack frames and updated the public generic error methods to all call directly into `newGenericError`.